newIDE/app/src/MainFrame/index.js

```diff
@@ -761,6 +761,42 @@ const MainFrame = (props: Props) => {
     [updateToolbar]
   );
 
+  const informEditorsResourceExternallyChanged = React.useCallback(
+    (resourceInfo: {| identifier: string |}) => {
+      ResourcesLoader.burstAllUrlsCache();
+      state.editorTabs.editors.forEach(editorTab => {
+        if (
+          editorTab.editorRef &&
+          editorTab.editorRef.editor &&
+          editorTab.editorRef.editor.onResourceExternallyChanged
+        ) {
+          // Each editor container has an accessible editor property.
+          // $FlowFixMe[not-a-function]
+          editorTab.editorRef.editor.onResourceExternallyChanged(resourceInfo);
+        }
+      });
+    },
+    [state.editorTabs]
+  );
+
+  React.useEffect(
```

Comment:
may I suggest moving this effect & callback in to a hook, so that you can do `useResourceWatcher(storageProvider, currentFileMetadata, editorTabs)` and it is in charge of the rest?

Reply:
Good idea, I'll change this